appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java

```diff
@@ -4861,24 +4738,40 @@ protected void postInvokeNoTx(EjbInvocation inv) {
         // No-op by default
     }
 
+    /**
+     * Checks if the exception is not null and if it is checked exception
+     * - uses ejb descriptor to resolve it.
+     *
+     * @param exception can be null
+     * @return true if the exception is not null and if it is not an unchecked exception
+     */
     protected boolean isApplicationException(Throwable exception) {
-        return !isSystemUncheckedException(exception);
+        return exception != null && !isSystemUncheckedException(exception);
```

Comment:
All calls of this method checked null first. The method should have the same "convention" as isSystemUncheckedException, so at least argument should be an exception at least. Null is not an exception and negation was not enough.
Bonus: I could remove all those null checks in calls.